### PR TITLE
CORCI-745 build: Save packages to Repository

### PR DESCRIPTION
**Pull request by JohnMalmberg**
* If the environment variables are set, save artifacts to the repositor
  servers.

* Get SLES/LEAP artifacts from the repository server if available.

Signed-off-by: John.E.Malmberg <john.e.malmberg@intel.com>

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -94,6 +94,11 @@ pipeline {
                             sh '''(cd /var/lib/mock/epel-7-x86_64/result/ &&
                                    cp -r . $OLDPWD/artifacts/centos7/)
                                   createrepo artifacts/centos7/'''
+                            publishToRepository product: 'slurm',
+                                                format: 'yum',
+                                                maturity: 'stable',
+                                                tech: 'el7',
```
Comment (brianjmurrell) [inline Jenkinsfile]: Seems an odd name for what is effectively `distro`.

Comment (JohnMalmberg) [reply Jenkinsfile]: That naming convention was based https://jfrog.com/whitepaper/best-practices-structuring-naming-artifactory-repositories/

**Comment by JohnMalmberg** [comment]
"You didn't add the Packaging Check stage: https://github.com/daos-stack/openpa/blob/1167afa5e4e73eaeef281402c8a7e03e3fd2ca95/Jenkinsfile#L70-L99 that keeps us honest about the packaging."

The packaging check is based on PR branch for that repository.  I would prefer not to reference that until it gets merged into master.

**Comment by brianjmurrell** [comment]
> The packaging check is based on PR branch for that repository. I would prefer not to reference that until it gets merged into master.

That it's in a PR doesn't make it any less of an example of adding a packaging check to any given project's `Jenkinsfile`.  Every project is going to have it added.  There is no reason this one can't be the first one to land with it.

File: Jenkinsfile [hunk 2]
```diff
@@ -67,15 +67,39 @@ pipeline {
                                       make rpmlint''',
                            returnStatus: true
                     }
-                }
+                } // stage('RPM Lint')
+                stage('Check Packaging') {
```
Comment (brianjmurrell) [inline Jenkinsfile]: I have been putting the packaging check inside the Lint block if for no other reason than to parallelize and get done more quickly.

File: packaging/Dockerfile.leap.42.3 [hunk 3]
```diff
@@ -32,8 +32,6 @@ RUN if ! grep "^#includedir /etc/sudoers.d" /etc/sudoers; then              \
     fi;                                                                     \
     echo "build ALL=(ALL) NOPASSWD: /usr/bin/build" > /etc/sudoers.d/build; \
     echo "build ALL=(ALL) NOPASSWD: /bin/rpm" >> /etc/sudoers.d/build;      \
```
Comment (brianjmurrell) [inline packaging/Dockerfile.leap.42.3]: Why is `/bin/rpm` needed on Leap 42.3 but not Leap 15?

File: packaging/Dockerfile.sles.12.3 [hunk 4]
```diff
@@ -32,8 +32,6 @@ RUN if ! grep "^#includedir /etc/sudoers.d" /etc/sudoers; then              \
     fi;                                                                     \
     echo "build ALL=(ALL) NOPASSWD: /usr/bin/build" > /etc/sudoers.d/build; \
     echo "build ALL=(ALL) NOPASSWD: /bin/rpm" >> /etc/sudoers.d/build;      \
```
Comment (brianjmurrell) [inline packaging/Dockerfile.sles.12.3]: Ditto.

File: Jenkinsfile (hunk 2 above)
Comment (JohnMalmberg) [reply Jenkinsfile]: I missed adding a parallel stage to that.
It does not look like it buys us much time though.
10 seconds for lint + 5 seconds for Check packaging.

File: packaging/Dockerfile.leap.42.3 (hunk 3 above)
Comment (JohnMalmberg) [reply packaging/Dockerfile.leap.42.3]: The Leap-15 dockerfile is not used for this.  The packaging repository will need to be updating before this is changed here.

The /bin/rpm is used for installing a GPG key needed for a repository only used for SLES 12.3 and Leap 42.3, but the makefile_packaging.mk is downloading and installing that key for all SLES-15/LEAP-15 also.

More likely the makefile_packaging.mk in the packaging repository should be fixed to not install the key when it is not needed.

**Pull request merged**